src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('button');
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const scoreBoard = document.querySelector('.game-score');
+const tableCells = document.querySelectorAll('td');
+const size = 4;
+let score = 0;
+let scoreNumber = [];
+
+const keyCode = {
+  left: 37,
+  up: 38,
+  right: 39,
+  down: 40,
+};
+
+let gameTable = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function isGameTableFull() {
+  for (let i = 0; i < size; i++) {
+    if (gameTable[i].includes(0)) {
+      return false;
+    }
+  }
+}
+
+function randomNumber() {
+  while (!isGameTableFull()) {
```

Comment:
Avoid using magic numbers in your code, such as the '4' in `Math.floor(Math.random() * 4)`. Consider using the 'size' variable instead.